src/gl.rs

```diff
@@ -107,386 +60,89 @@ fn calculate_length(width: GLsizei, height: GLsizei, format: GLenum, pixel_type:
-#[inline]
-pub fn gen_queries(n: GLsizei) -> Vec<GLuint> {
-    unsafe {
-        let mut result = vec![0; n as usize];
-        ffi::GenQueries(n, result.as_mut_ptr());
-        return result;
-    }
-}
-
-#[cfg(not(target_os="android"))]
-#[inline]
-pub fn begin_query(target: GLenum, id: GLuint) {
-    unsafe {
-        ffi::BeginQuery(target, id);
-    }
-}
-
-#[cfg(not(target_os="android"))]
-#[inline]
-pub fn end_query(target: GLenum) {
-    unsafe {
-        ffi::EndQuery(target);
-    }
-}
-
-#[cfg(not(target_os="android"))]
-#[inline]
-pub fn query_counter(id: GLuint, target: GLenum) {
-    unsafe {
-        ffi::QueryCounter(id, target);
-    }
-}
-
-#[cfg(not(target_os="android"))]
-#[inline]
-pub fn get_query_object_iv(id: GLuint, pname: GLenum) -> i32 {
-    unsafe {
-        let mut result = 0;
-        ffi::GetQueryObjectiv(id, pname, &mut result);
-        result
-    }
-}
-
-#[cfg(not(target_os="android"))]
-#[inline]
-pub fn get_query_object_uiv(id: GLuint, pname: GLenum) -> u32 {
-    unsafe {
-        let mut result = 0;
-        ffi::GetQueryObjectuiv(id, pname, &mut result);
-        result
-    }
-}
-
-#[cfg(not(target_os="android"))]
-#[inline]
-pub fn get_query_object_i64v(id: GLuint, pname: GLenum) -> i64 {
-    unsafe {
-        let mut result = 0;
-        ffi::GetQueryObjecti64v(id, pname, &mut result);
-        result
-    }
-}
-
-#[cfg(not(target_os="android"))]
-#[inline]
-pub fn get_query_object_ui64v(id: GLuint, pname: GLenum) -> u64 {
-    unsafe {
-        let mut result = 0;
-        ffi::GetQueryObjectui64v(id, pname, &mut result);
-        result
-    }
-}
-
-#[cfg(not(target_os="android"))]
-#[inline]
-pub fn delete_queries(queries: &[GLuint]) {
-    unsafe {
-        ffi::DeleteQueries(queries.len() as GLsizei, queries.as_ptr());
-    }
-}
-
-#[inline]
-pub fn delete_vertex_arrays(vertex_arrays: &[GLuint]) {
-    unsafe {
-        ffi::DeleteVertexArrays(vertex_arrays.len() as GLsizei, vertex_arrays.as_ptr());
-    }
-}
-
-#[inline]
-pub fn delete_buffers(buffers: &[GLuint]) {
-    unsafe {
-        ffi::DeleteBuffers(buffers.len() as GLsizei, buffers.as_ptr());
-    }
-}
-
-#[inline]
-pub fn delete_renderbuffers(renderbuffers: &[GLuint]) {
-    unsafe {
-        ffi::DeleteRenderbuffers(renderbuffers.len() as GLsizei, renderbuffers.as_ptr());
-    }
-}
-
-#[inline]
-pub fn delete_framebuffers(framebuffers: &[GLuint]) {
-    unsafe {
-        ffi::DeleteFramebuffers(framebuffers.len() as GLsizei, framebuffers.as_ptr());
-    }
-}
-
-// NB: The name of this function is wrong, it's here for compatibility reasons,
-// but should be removed.
-#[inline]
-pub fn delete_frame_buffers(framebuffers: &[GLuint]) {
-    delete_framebuffers(framebuffers);
-}
-
-#[inline]
-pub fn delete_textures(textures: &[GLuint]) {
-    unsafe {
-        ffi::DeleteTextures(textures.len() as GLsizei, textures.as_ptr());
-    }
-}
-
-#[inline]
-pub fn framebuffer_renderbuffer(target: GLenum,
+pub trait Gl : Send + Sync
```

Comment:
Do you mean, we have to recreate trait GL object before wrapper.new_context()? To re-create  trait GL object in RenderBackend thread, we need to get gl load function. Do we reuse the load function? But if we reuse the load function, Renerer and RenderBackend calls same library's functions. It is virtually same to share GL trait between Renerer and RenderBackend. How do you think about it?